engine/src/main/java/io/camunda/zeebe/engine/Engine.java

```diff
@@ -65,12 +66,15 @@ public Engine(final TypedRecordProcessorFactory typedRecordProcessorFactory) {
 
   @Override
   public void init(final RecordProcessorContext recordProcessorContext) {
+    final var zeebeDb = recordProcessorContext.getZeebeDb();
     zeebeState =
         new ZeebeDbState(
             recordProcessorContext.getPartitionId(),
-            recordProcessorContext.getZeebeDb(),
+            zeebeDb,
             recordProcessorContext.getTransactionContext(),
             recordProcessorContext.getKeyGenerator());
+    final var scheduledTaskDbState = new ScheduledTaskDbState(zeebeDb, zeebeDb.createContext());
```

Comment:
🔧 I don't necessarily mind a specialized class like `ScheduledTaskDbState`, but an Immutable state interface already exists called `ZeebeState`. It could be used instead:
```java
final ZeebeState scheduledTaskState = new ZeebeDbState(zeebeDb, zeebeDb.createContext());
```

💭 Clearly, this has one major difference: It provides all read-only state classes instead of only the message state. Would that cause any issues? Well, only when the read-only methods are used that aren't actually read-only 🤷 IMO, it's fine to expose those for now, as we will need to work to make them completely read-only in future iterations.

Reply:
:+1: We considered this but consciously chose to introduce a specialized state holder instead of using `ZeebeState` just because it seems wasteful to build state objects that won't be used.

Reply:
Okay, that makes sense. It also helps in the migration to splitting the actor for all scheduled tasks eventually.